cdap-security/src/main/java/io/cdap/cdap/security/authorization/DefaultAccessEnforcer.java

```diff
@@ -61,140 +60,134 @@
   @Nullable
   private final Principal masterUser;
   private final int logTimeTakenAsWarn;
+  private final AccessEnforcer internalAccessEnforcer;
+  private final boolean internalAuthEnabled;
 
   @Inject
   DefaultAccessEnforcer(CConfiguration cConf, SConfiguration sConf,
-                        AccessControllerInstantiator accessControllerInstantiator) {
+                        AccessControllerInstantiator accessControllerInstantiator,
+                        @Named(INTERNAL_ACCESS_ENFORCER) AccessEnforcer internalAccessEnforcer) {
     super(cConf);
     this.sConf = sConf;
     this.accessControllerInstantiator = accessControllerInstantiator;
     String masterUserName = AuthorizationUtil.getEffectiveMasterUser(cConf);
     this.masterUser = masterUserName == null ? null : new Principal(masterUserName, Principal.PrincipalType.USER);
     this.logTimeTakenAsWarn = cConf.getInt(Constants.Security.Authorization.EXTENSION_OPERATION_TIME_WARN_THRESHOLD);
+    this.internalAccessEnforcer = internalAccessEnforcer;
+    this.internalAuthEnabled = SecurityUtil.isInternalAuthEnabled(cConf);
   }
 
   @Override
   public void enforce(EntityId entity, Principal principal, Set<? extends Permission> permissions)
     throws AccessException {
-    doEnforce(entity, principal, permissions);
-  }
-
-  @Override
-  public void enforceOnParent(EntityType entityType, EntityId parentId, Principal principal, Permission permission)
-    throws AccessException {
-    doEnforce(entityType, parentId, principal, permission);
-  }
-
-  @Override
-  public Set<? extends EntityId> isVisible(Set<? extends EntityId> entityIds, Principal principal)
-    throws AccessException {
-    // TODO CDAP-17772: Temporary short-circuit, implement internal access enforcement
-    if (principal.getFullCredential() != null && principal.getFullCredential().getType()
+    if (internalAuthEnabled && principal.getFullCredential() != null && principal.getFullCredential().getType()
```

Comment:
you don't need equals for enums. `==` works fine

Reply:
Fixed